### ISSUE TITLE
Incompatibility with newest "importlib_resources" backport of "importlib.resources"

### ISSUE THREAD

**Issue by SupImDos** [Nuitka/Nuitka#1274]
## Description

First off, thank you for all your hard work :smiley:

I have found that a `--onefile` / `--standalone` application using the package [`jsonschema`](https://pypi.org/project/jsonschema/) no longer works, since `jsonschema` v4.x.x.

I have narrowed it down to the `jsonschema._utils.load_schema` function, which appears to have changed from using `pkgutil` to `importlib.resources` / `importlib_resources` in its v3 -> v4 update. This seems to have broken its functionality with Nuitka.

In `jsonschema` 3.2.0 (working), the `load_schema` function used `pkgutil`
```python3
def load_schema(name):
    """
    Load a schema from ./schemas/``name``.json and return it.
    """

    data = pkgutil.get_data("jsonschema", "schemas/{0}.json".format(name))
    return json.loads(data.decode("utf-8"))
```

In `jsonschema` 4.2.1 (no longer working), it appears the `load_schema` function now uses `importlib_resources`
```python3
if sys.version_info >= (3, 9):  # pragma: no cover
    import importlib.resources as resources
else:  # pragma: no cover
    import importlib_resources as resources

...

def load_schema(name):
    """
    Load a schema from ./schemas/``name``.json and return it.
    """

    path = resources.files(__package__).joinpath(f"schemas/{name}.json")
    data = path.read_text(encoding="utf-8")
    return json.loads(data)
```

## Error Logs
```console
$ ./example

Traceback (most recent call last):
  File "/tmp/.mount_examplWYktsX/example.py", line 1, in <module>
    import jsonschema
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "/tmp/.mount_examplWYktsX/jsonschema/__init__.py", line 29, in <module jsonschema>
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "/tmp/.mount_examplWYktsX/jsonschema/validators.py", line 349, in <module jsonschema.validators>
  File "/tmp/.mount_examplWYktsX/jsonschema/_utils.py", line 61, in load_schema
  File "/tmp/.mount_examplWYktsX/importlib_resources/abc.py", line 73, in read_text
  File "/tmp/.mount_examplWYktsX/importlib_resources/_adapters.py", line 141, in open
FileNotFoundError: Can't open orphan path
```

## Nuitka Version:
Tried on both latest version, and development branch version.
  * `v0.6.17.6`
  * `develop` branch (v0.6.18rc7)

## Operating System
  * Ubuntu 20.04
  * x86_64
  * Linux 5.8.0-59-generic

## Installation Method:
  * `pip` (`poetry` with virtual environment and `pyproject.toml`

## Example
`example.py`
```python3
import jsonschema
print(jsonschema)
```

`pyproject.toml`
```toml
[tool.poetry]
name = "example"
version = "0.1.0"
description = ""
authors = ["..."]

[tool.poetry.dependencies]
python = "^3.8"
jsonschema = "^4.2.1"
Nuitka = "^0.6.17"

[tool.poetry.dev-dependencies]

[build-system]
requires = ["poetry-core>=1.0.0"]
build-backend = "poetry.core.masonry.api"
```

`nuitka.sh`
```bash
#!/bin/bash
python3 -m nuitka  \
    example.py     \
    -o example     \
    --onefile
```

**Comment by kayhayen** [comment]
It seems `importlib_resources` is supposed to be working as of 0.6.1 which is a long time away, but maybe it is the one that changed. And I also couldn't reproduce this with 3.9 and only with 3.8, which is using that backport.

**Comment by kayhayen** [comment]
This must be new:

```
        def _file_reader(spec):
            try:
                path = pathlib.Path(self.path)
            except TypeError:
                return None
            if path.exists():
                return readers.FileReader(self)
```

In this, self.path is `.../Mini.dist/jsonschema/__init__.py` and that doesn't exist, and therefore the package files do not exist in the file system. Nuitka is puting into `__spec__.origin` and `__file__` and the like, paths that do not exist, so people can build relative paths using them. Check if it exists, is against the rules. Replacing it with `True` in the condition will make it work.

**Comment by kayhayen** [comment]
The fix is on https://nuitka.net/doc/factory.html and will appear in a hotfix soon.

**Comment by SupImDos** [comment]
I have confirmed that this issue is now fixed for me in `v0.6.17.7`. Thank you!

**Comment by kayhayen** [comment]
Part of the hotfix indeed. Thanks for your report.
